### ISSUE TITLE
Allow keyboard navigation from highlight in host frame to associated annotation in sidebar

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/client#4741]
> When users are using the screen reader to read through a document, they are told an annotation exists when passing over one. However, they cannot jump to the annotation and read it. It is worth noting here that using the annotation pane to read annotations enables users to jump to the point in the document that was annotated.

Notes: 

* May be subject to browser differences WRT focus wrangling
* There may be some need to announce a keyboard shortcut here, too
* There may be an orthogonal need to validate standalone documentation for how to use H with keyboard/screen reader

**Comment by robertknight** [comment]
We already have a "Show" button that appears in the toolbar when already-highlighted text is selected, so one possibility is that when this button is triggered, we not only filter annotations in the sidebar (as we do now), but also transfer keyboard focus to the sidebar and to the annotation card for the first annotation in the selected set. Assuming the user has a way to select text with the keyboard, this provides a keyboard-only way to navigate from the highlight to the annotation in the sidebar. Something that we'd have to look into is how this affects the process of getting back to the place in the document where the user's selection was previously.

In addition to focus transfer when clicking "Show", we'd also need to announce the "Show" shortcut. Perhaps the solution we land on for https://github.com/hypothesis/client/issues/4742 can be extended to include this.

**Comment by robertknight** [comment]
One issue I ran into while implementing this is that the client is currently a bit trigger happy when it comes to selecting annotations. The selection in the sidebar is changed not only when clicking on highlighted text, but also whenever the cursor is released inside a highlight. The "Show" action in the toolbar currently just triggers an update of the selection. If we're going to make it also move focus to the sidebar, we need to make sure that doesn't happen when eg. making a text selection that happens to end inside a highlight.

**Comment by robertknight** [comment]
> We already have a "Show" button that appears in the toolbar when already-highlighted text is selected, so one possibility is that when this button is triggered, we not only filter annotations in the sidebar (as we do now), but also transfer keyboard focus to the sidebar and to the annotation card for the first annotation in the selected set

I have this working locally. It works as intended in Chrome and Firefox. In Safari there is a complication that the sidebar cannot request focus until the user has interacted with it at least once since the top-level frame loaded. Prior to that `window.focus()` calls inside the sidebar are ignored, and if the host frame attempts to call `focus()` on the iframe element, the frame element is focused, but not the window it contains.
